File: apps/java/src/main/java/tdd/fizzbuzz/domain/model/FizzBuzzValue.java
package tdd.fizzbuzz.domain.model;

import java.util.Objects;

public final class FizzBuzzValue {

    private final int number;
    private final String value;

    public FizzBuzzValue(int number, String value) {
        this.number = number;
        this.value = value;
    }

    public int getNumber() {
        return number;
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return number + ":" + value;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null || getClass() != obj.getClass()) {
            return false;
        }
        FizzBuzzValue that = (FizzBuzzValue) obj;
        return number == that.number && Objects.equals(value, that.value);
    }

    @Override
    public int hashCode() {
        return Objects.hash(number, value);
    }
}

File: apps/php/src/Domain/Model/FizzBuzzValue.php
<?php

namespace App\Domain\Model;

use App\Domain\FizzBuzzException;

final class FizzBuzzValue
{
    public function __construct(
        private readonly int $number,
        private readonly string $value,
    ) {
        if ($number < 0) {
            throw new FizzBuzzException('値は正の値のみ許可します');
        }
    }

    public function getNumber(): int
    {
        return $this->number;
    }

    public function getValue(): string
    {
        return $this->value;
    }

    public function equals(self $other): bool
    {
        return $this->number === $other->number
            && $this->value === $other->value;
    }

    public function __toString(): string
    {
        return $this->value;
    }
}
